app/src/main/java/org/mozilla/fenix/components/metrics/Metrics.kt

```diff
@@ -71,6 +74,29 @@ sealed class Event {
     object CustomTabsActionTapped : Event()
     object CustomTabsMenuOpened : Event()
 
+    data class PreferenceToggled(val preferenceKey: String, val enabled: Boolean, val context: Context) : Event() {
+        private val switchPreferenceTelemetryAllowList = listOf(
+            context.getString(R.string.pref_key_leakcanary),
```

Comment:
So every single preference toggle needs to grab all of these keys from resources? I rather wish there was a way to do this only once and remember the set or list, but this will do for now.